Plants

BoogieMonster1O1 opened the pull request:
**Version:-** Snapshot 1.0.0+alpha.7


### Changes
* Plants


#### Additional Context
this supports a few stuff that plain blocks don't support 😛 
```make
 On branch feature/plants
 Changes to be committed:
	new file:   src/fabric-1.16.2/plant.definition.yaml
	modified:   src/fabric-1.16.2/templates/modbase/client.java.ftl
	modified:   src/fabric-1.16.2/templates/modbase/mod.java.ftl
	new file:   src/fabric-1.16.2/templates/plant.java.ftl
```

BoogieMonster1O1 committed:
Added plants

 On branch feature/plants
 Changes to be committed:
	new file:   src/fabric-1.16.2/plant.definition.yaml
	modified:   src/fabric-1.16.2/templates/modbase/client.java.ftl
	modified:   src/fabric-1.16.2/templates/modbase/mod.java.ftl
	new file:   src/fabric-1.16.2/templates/plant.java.ftl

BoogieMonster1O1 committed:
Bump version

 Changes to be committed:
	modified:   src/plugin.json

NOYBss commented on src/fabric-1.16.2/templates/plant.java.ftl:
```diff
@@ -0,0 +1,218 @@
+        .nonOpaque()
+		<#if data.emissiveRendering>
+                .emissiveLighting(${name}Block::always)
+        </#if>
```
Do we still need this since it also overrides emissive rendering below
Also it will probably give a build error because ${name}Block won't exist

BoogieMonster1O1 replied:
didn't realize that, thank you for pointing it out.
now fixed

BoogieMonster1O1 committed:
Removed emissive rendering from settings

 Changes to be committed:
	modified:   src/fabric-1.16.2/templates/plant.java.ftl

NOYBss commented on src/fabric-1.16.2/plant.definition.yaml:
```diff
@@ -0,0 +1,26 @@
+    condition: "plantType %= normal"
+    variables: "variant= "
+    name: "@MODASSETSROOT/blockstates/@registryname.json"
+  - template: json/block_states_gplant.json.ftl
```
Fabric Generator doesn't have this file

BoogieMonster1O1 replied:
fixed, added it

BoogieMonster1O1 committed:
Added a missing model file

 Changes to be committed:
	new file:   src/fabric-1.16.2/templates/json/block_states_gplant.json.ftl

NOYBss reviewed (approved):
Looks good to me

Goldorion committed:
Rechange the plugin version

We are working on the alpha 6, not the alpha 7.

Goldorion commented:
I just changed the plugin version inside the plugin.json because we didn't release the alpha #6 currently, as we are working on it.
